src/Raven.Server/Documents/Handlers/Admin/Processors/Indexes/AdminIndexHandlerProcessorForJavaScriptPut.cs

```diff
@@ -25,22 +24,19 @@ public AdminIndexHandlerProcessorForJavaScriptPut([NotNull] DatabaseRequestHandl
 
     protected override async ValueTask HandleIndexesFromLegacyReplicationAsync()
     {
-        if (HttpContext.Features.Get<IHttpAuthenticationFeature>() is RavenServer.AuthenticateConnection feature &&
-            feature.CanAccess(RequestHandler.DatabaseName, requireAdmin: true, requireWrite: true) == false)
+        if (AuthenticateConnection != null &&
+            AuthenticateConnection.CanAccess(RequestHandler.DatabaseName, requireAdmin: true, requireWrite: true) == false)
         {
             throw new UnauthorizedAccessException("Deploying indexes from legacy replication requires admin privileges.");
         }
 
+        var options = new DatabaseSmugglerOptionsServerSide(RequestHandler.GetAuthorizationStatusForSmuggler(RequestHandler.DatabaseName));
         using (ContextPool.AllocateOperationContext(out JsonOperationContext jsonOperationContext))
         await using (var stream = new ArrayStream(RequestHandler.RequestBodyStream(), nameof(DatabaseItemType.Indexes)))
-        using (var source = new StreamSource(stream, jsonOperationContext, RequestHandler.DatabaseName))
+        using (var source = new StreamSource(stream, jsonOperationContext, RequestHandler.DatabaseName, options))
         {
             var destination = RequestHandler.Database.Smuggler.CreateDestination();
-            var options = new DatabaseSmugglerOptionsServerSide
-            {
-                OperateOnTypes = DatabaseItemType.Indexes
-            };
-
+            options.OperateOnTypes = DatabaseItemType.Indexes;
```

Comment:
Let's initialize this when creating the options:
```
var options = new DatabaseSmugglerOptionsServerSide(RequestHandler.GetAuthorizationStatusForSmuggler(RequestHandler.DatabaseName))
{
      OperateOnTypes = DatabaseItemType.Indexes
}
```

Reply:
Done